addons/mail/static/src/discuss/core/public_web/thread_model_patch.js

```diff
@@ -115,18 +115,22 @@ patch(Thread.prototype, {
         }
         const limit = 30;
         const data = await rpc("/discuss/channel/sub_channel/fetch", {
-            before: this.lastSubChannelLoaded,
+            before: this.lastSubChannelLoaded?.id,
             limit,
             parent_channel_id: this.id,
             search_term: searchTerm,
         });
-        const { Thread: subChannels = [] } = this.store.insert(data, { html: true });
+        const { Thread: threads = [] } = this.store.insert(data, { html: true });
         if (searchTerm) {
             // Ignore holes in the sub-channel list that may arise when
             // searching for a specific term.
             return;
         }
-        this.lastSubChannelLoaded = subChannels.at(-1)?.id;
+        const subChannels = threads.filter((thread) => this.eq(thread.parent_channel_id));
+        this.lastSubChannelLoaded = subChannels.reduce(
+            (min, channel) => (!min || channel.id < min.id ? channel : min),
+            this.lastSubChannelLoaded
+        );
```

Comment:
I feel like `reduce` is harder to read and understand and is not that much shorter.

```suggestion
        let min;
        for (const channel of subChannels) {
            if (!min || channel.id <min.id) {
                min = channel;
            }
        }

Reply:
Isn't native code always preferable to doing things manually?
`reduce` isn't really hard to read once we get used to it.

Reply:
IMO using reduce complicates straightforward tasks for the sake of saving a couple of lines. I wouldn’t sacrifice readability for brevity. However, since this is a matter of preference, I’ll leave it up to you to decide whether to apply my suggestion. 🙂

Reply:
I don't mind too much either way, @alexkuhn what do you think?

Reply:
Pretty indifferent, they both work good: `.reduce()` syntax is shorter and has less side-effect than the `for of` variant, but the `for-of` variant is easier to understand and maintain for any developer.

If you really want me to choose... Maybe the `.reduce()`, just because I can better see its block and ignore it, and there's LOCs so that means I have to scroll less to have a macro-view of the file.

Reply:
Then I guess we'll leave it like this, can you check for security and merging?